src/common/snippets/include/snippets/lowered/loop_manager.hpp

```diff
@@ -154,56 +214,139 @@ class LinearIR::LoopManager {
+     *              all decompositions should be call on this pipeline
+     * @param new_expr_begin the first expression iterator
+     * @param new_expr_end the next iterator after the last expression
+     * @param decomposed_expr the expression that is decomposed into several other exprs
+     * @param loop_id the new expression will be marked by this loop ID
+     * @param new_entries new input loop ports
+     * @param new_exits new output loop ports
+     */
     void expression_replacement(constExprIt new_expr_begin, constExprIt new_expr_end, const ExpressionPtr& decomposed_expr,
                                 size_t loop_id, const std::vector<ExpressionPort>& new_entries, const std::vector<ExpressionPort>& exits);
+    /**
+     * @brief Find iterators of first entry loop point and last exit point (bounds of Loop) when explicit Loop exprs haven't been inserted in LinearIR
+     *        Note: This method cannot be called when explicit loop exprs are inserted since Loop bounds are defined by LoopBegin and LoopEnd after the insertion!
+     * @param linear_ir linear IR
+     * @param loop_id target Loop ID
+     * @param loop_begin_pos the ref of first expression iterator of the Loop
+     * @param loop_end_pos the ref of next iterator after the last expression
+     */
+    void get_loop_bounds(const LinearIR& linear_ir, size_t loop_id, LinearIR::constExprIt& loop_begin_pos, LinearIR::constExprIt& loop_end_pos) const;
+    /**
+     * @brief Find iterators of first entry loop point and last exit point (bounds of Loop) when explicit Loop exprs haven't been inserted in LinearIR
+     *        Note: This method cannot be called when explicit loop exprs are inserted since Loop bounds are defined by LoopBegin and LoopEnd after the insertion!
+     * @param linear_ir linear IR
+     * @param loop_id target Loop ID
+     * @param entries input loop ports
+     * @param exits output loop ports
+     * @param loop_begin_pos the ref of first expression iterator of the Loop
+     * @param loop_end_pos the ref of next iterator after the last expression
+     */
+    static void get_loop_bounds(const LinearIR& linear_ir, size_t loop_id, const std::vector<LoopPort>& entries, const std::vector<LoopPort>& exits,
+                                LinearIR::constExprIt& loop_begin_pos, LinearIR::constExprIt& loop_end_pos);
+    /**
+     * @brief Finds iterators of LoopBegin and LoopEnd when explicit Loop exprs have been already inserted in LinearIR
+     * @param linear_ir linear IR
+     * @param loop_id target Loop ID
+     * @param loop_begin_pos the ref of first expression iterator of the Loop
+     * @param loop_end_pos the ref of next iterator after the last expression
+     */
+    void get_explicit_loop_bounds(const LinearIR& linear_ir, size_t loop_id, LinearIR::constExprIt& loop_begin_pos, LinearIR::constExprIt& loop_end_pos) const;
```

Comment:
We should try to avoid creating similar methods that valid only in certain situations. It's like having two separate functions: `print_char` and `print_float` instead of one `print` that could work with all formats.
Why can't `get_loop_bounds` return iterators to LoopBegin and LoopEnd?